### ISSUE TITLE
nonet flag isn't applied automatically in all cases - PCL Reference Assemblies not installed

### ISSUE THREAD

**Issue by shearer12345** [AtomicGameEngine/AtomicGameEngine#1544]
The linux build instructions say that
> The --nonet flag will automatically be applied if Visual Studio or mono isn't installed on the system

My build (`sh Build_AtomicEditor.sh`) errors with
`/usr/lib/mono/4.5/Microsoft.Common.targets: error : PCL Reference Assemblies not installed.`

 as mono wasn't installed. 

It seems that it isn't detection that mono isn't installed, and is trying to build with C# features.

* Installing mono ([as detailed](https://github.com/AtomicGameEngine/AtomicGameEngine/wiki/Detailed-instructions-for-building-on-Linux)) fixed this
* Adding `--nonet` manually also works (after removing mono)

Happy to look more closely if you can direct to where the detection of mono's installation happens

## System details:
Ubuntu 16.10 64 bit

## Logs
[Atomic.log is attached (as .txt)](https://github.com/AtomicGameEngine/AtomicGameEngine/files/1026615/Atomic.log.txt)

tail of console follows
```
Errors:

/home/shearer/code/AtomicGameEngine/Artifacts/AtomicNET/Build/AtomicNET.sln (default targets) ->
(Build target) ->
/home/shearer/code/AtomicGameEngine/Artifacts/AtomicNET/Build/AtomicNET/AtomicNET.csproj (default targets) ->
/usr/lib/mono/4.5/Microsoft.Common.targets (GetReferenceAssemblyPaths target) ->

	/usr/lib/mono/4.5/Microsoft.Common.targets: error : PCL Reference Assemblies not installed.

	 57 Warning(s)
	 1 Error(s)

Time Elapsed 00:00:01.9028370

Compilation Command: -c mono "/home/shearer/code/AtomicGameEngine/Build/Managed/nuget/nuget.exe" restore "/home/shearer/code/AtomicGameEngine/Artifacts/AtomicNET/Build/AtomicNET.sln" && "xbuild" "/home/shearer/code/AtomicGameEngine/Artifacts/AtomicNET/Build/AtomicNET.sln" /p:Platform="Any CPU" /p:Configuration=Release

NETBuild Error for solution: /home/shearer/code/AtomicGameEngine/Script/AtomicNET/AtomicNETProject.json
/home/shearer/code/AtomicGameEngine/Build/node_modules/jake/lib/api.js:340
        throw errObj;
        ^

Error: Process exited with error.
    at api.fail (/home/shearer/code/AtomicGameEngine/Build/node_modules/jake/lib/api.js:336:18)
    at Exec.<anonymous> (/home/shearer/code/AtomicGameEngine/Build/node_modules/jake/lib/utils/index.js:124:9)
    at emitTwo (events.js:106:13)
    at Exec.emit (events.js:194:7)
    at ChildProcess.<anonymous> (/home/shearer/code/AtomicGameEngine/Build/node_modules/jake/lib/utils/index.js:227:20)
    at emitTwo (events.js:106:13)
    at ChildProcess.emit (events.js:194:7)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:215:12)
```

**Comment by JoshEngebretson** [comment]
Thanks for the report, we're looking for xbuild here and turn on ```nonet``` if not found, probably needs a better check: https://github.com/AtomicGameEngine/AtomicGameEngine/blob/master/Build/Scripts/BuildConfig.js#L36

**Comment by JimMarlowe** [comment]
The code from the resulting PR #1547 breaks my ability to build the C# subsystem.  My system has an existing mono installation that did(does) function properly. The root of the problem is, it does not have a directory at /usr/bin/mono, it instead has an executable (a link to mono-sgen), and now very silently will not include the ability to create new C# projects.
```
JimMarlowe@marlowe-09 ~ $ inxi -Fxz
System:    Host: marlowe-09 Kernel: 3.13.0-24-generic x86_64 (64 bit, gcc: 4.8.2) 
           Desktop: Gnome  (Gtk 3.10.8) Distro: Linux Mint 17 Qiana

JimMarlowe@marlowe-09 ~ $ which mono
/usr/bin/mono
JimMarlowe@marlowe-09 ~ $ ls -ltr /usr/bin/mono
lrwxrwxrwx 1 root root 9 May 25 02:39 /usr/bin/mono -> mono-sgen
JimMarlowe@marlowe-09 ~ $  ls -ltr /usr/bin/mono-sgen
-rwxr-xr-x 1 root root 3785528 May 25 02:39 /usr/bin/mono-sgen
JimMarlowe@marlowe-09 ~ $ file /usr/bin/mono-sgen
/usr/bin/mono-sgen: ELF 64-bit LSB  executable, x86-64, version 1 (SYSV), dynamically linked (uses shared libs), for GNU/Linux 2.6.26, BuildID[sha1]=e299f70ce46e18f6d2dd58cadc1355223aead57d, stripped
```

**Comment by JoshEngebretson** [comment]
Ok, so what probably needs to happen is a check for whether a mono directory or file exists?  Maybe we can use ```which``` like is done for macOS immediately above?

https://github.com/AtomicGameEngine/AtomicGameEngine/blob/master/Build/Scripts/BuildConfig.js#L48

**Comment by JoshEngebretson** [comment]
Closed in #1575

**Comment by shearer12345** [comment]
@JimMarlowe Sorry about breaking your build. Thanks for the fix.
